class.jetpack.php

```diff
@@ -468,6 +468,12 @@ private function __construct() {
 		add_action( 'init', array( $this, 'deprecated_hooks' ) );
 
 		/*
+		 * Enable enhanced handling of previewing sites in Calypso
+		 */
+		require_once JETPACK__PLUGIN_DIR . '_inc/lib/class.jetpack-iframe-embed.php';
```

Comment:
Is it necessary to always include this file which is the case with this line here? If it is, then it should be placed in `jetpack.php` along with the other file loading routines.

If it's not, which I think must be the case due to `is_embedding_in_iframe()` then the logic should be extracted so it's in another file we always load, like this `class.jetpack.php` and we can avoid loading another file every time the site loads and we do it only when it's needed.

Reply:
> then the logic should be extracted so it's in another file we always load

@eliorivero Doesn't that make it a completely manual process to keep in sync? If the loading logic is contained in the class & consulted during the `init` call, keeping the file in sync is dead simple.